start.sh

```diff
@@ -0,0 +1,147 @@
+
+  if [[ "$DISTRO_ID" == "ubuntu" ]]; then
+    install_docker_ubuntu
+  elif [[ "$DISTRO_ID" == "arch" ]]; then
+    install_docker_arch
+  else
+    echo "[WARNING] Your distro is '$DISTRO_ID', which isn't supported by this script."
+    echo "          Please follow Docker's official instructions at:"
+    echo "          https://docs.docker.com/get-docker/"
+    exit 1
+  fi
+
+  # Add current user to docker group
+  echo "[INFO] Adding '$USER' to the 'docker' group..."
+  sudo groupadd docker 2> /dev/null || true
+  sudo usermod -aG docker "$USER"
+
+  echo "[INFO] Docker installation complete."
+  echo "      You may need to log out and log back in (or restart your shell) for the group"
+  echo "      membership to take effect. Please do that, then re-run this script."
+  exit 0
+else
+  echo "[OK] Docker is already installed."
+fi
+
+# Double-check Docker is running properly
+if ! docker ps &> /dev/null; then
+  echo "[ERROR] Docker is installed but not running. Please start Docker (e.g., 'sudo systemctl start docker')."
+  exit 1
+fi
+
+###############################################################################
+# Check if the Docker image already exists
+###############################################################################
+# assume that Dockerfile is in the same directory
+echo "[INFO] Checking if the Docker image '$IMAGE_NAME' exists..."
+
+IMAGE_EXISTS=$(docker images -q "$IMAGE_NAME")
+
+if [ -z "$IMAGE_EXISTS" ]; then
+    echo "[INFO] '$IMAGE_NAME' not found. Building image from Dockerfile..."
+    
+    # Build the image
+    docker build -t "$IMAGE_NAME" "$DOCKERFILE_DIR"
+else
+    echo "[OK] Docker image '$IMAGE_NAME' already exists."
+fi
+
+###############################################################################
+# dockerfile hash
+###############################################################################
+# 1. Compute current hash of Dockerfile
+CURRENT_HASH="$(md5sum "$DOCKERFILE_PATH" | awk '{print $1}')"
+
+# 2. Compare with stored hash
+if [ -f "$HASH_FILE" ]; then
+    PREV_HASH="$(cat "$HASH_FILE")"
+    if [ "$CURRENT_HASH" != "$PREV_HASH" ]; then
+        echo "[INFO] Dockerfile changed. Rebuilding the image '$IMAGE_NAME'..."
+        docker build -t "$IMAGE_NAME" "$DOCKERFILE_DIR" || exit 1
+        echo "$CURRENT_HASH" > "$HASH_FILE"
+    else
+        echo "[INFO] Dockerfile unchanged. Using existing image '$IMAGE_NAME'."
+    fi
+else
+    echo "[INFO] No cached Dockerfile hash found. Building image '$IMAGE_NAME'..."
+    docker build -t "$IMAGE_NAME" "$DOCKERFILE_DIR" || exit 1
+    echo "$CURRENT_HASH" > "$HASH_FILE"
+fi
+
+###############################################################################
+# Run the container with:
+#    - USB passthrough 
+#    - Current directory mounted at /Embedded-Sharepoint
+#    - Interactive shell
+###############################################################################
+echo "[INFO] Running the Docker container ..."
```

Comment:
Maybe gate this with an if with a runtime arg so that people just using it for docker setup don't jump into a container shell they don't want. Not critical though.

Reply:
It should not be considered a setup script, it should be considered a start script for cli users. VS code users should not run it, the docker container extension handles everything for them.

The VS Code workflow:
Install Dev Container extension (if not yet installed)
Clone repo (if not yet cloned)
Open directory -> Extension gives initialization pop-up
At this point, the _Extension_ will actually handle setting up docker if it has not been installed on the system
It starts up the session as normal

CLI workflow:
run start.sh
if docker is not installed, it installs docker
otherwise, it starts up the session as normal

Reply:
Has this been tested on all probable deployment targets (vscode workflow)? On a clean Mac env like a week ago, I had an issue with docker not being preinstalled, and then also docker daemon socket issues.